### PR TITLE
Add one HPI dialog into babyMEG plugin. The basic function for this is t...

### PR DESCRIPTION
**Pull request by liminsun**
...o load the digitized points from Polhemus and further transfer those points to rthpi plugin. Rthpi will use those points to compute the dev_to_head transform matrix.

This is changed by Limin Sun.  Chiran changed codes in rt_hpi. So , please merge both.

The change of log2 to log10 is because log2 is not compiled in windows at the moment.

### CODE REVIEW

File: MNE/utils/mp/adaptivemp.cpp [hunk 1]
```diff
@@ -209,7 +209,7 @@ QList<QList<GaborAtom> > AdaptiveMp::matching_pursuit(MatrixXd signal, qint32 ma
         k = 0;
         s = sample_count;
         p = floor(sample_count / 2);
-        j = floor(log2(sample_count));//log(sample_count) / log(2));
+        j = floor(log10(sample_count));//log(sample_count) / log(2));
```
Comment (chdinh) [inline MNE/utils/mp/adaptivemp.cpp]: Is there a specific reason to change this line?

File: applications/mne_x/plugins/babymeg/FormFiles/babymeghpidgl.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+void babymeghpidgl::CancelProc()
+{
+
+}
+
+void babymeghpidgl::OKProc(QAbstractButton *b)
+{
+    qDebug()<<"Clicked group button";
+
+    auto s(b->text());
```
Comment (chdinh) [inline applications/mne_x/plugins/babymeg/FormFiles/babymeghpidgl.cpp]: please don't use the auto variable

File: applications/mne_x/plugins/rthpi/rthpi.cpp [hunk 3]
```diff
@@ -254,29 +254,24 @@ void RtHpi::run()
...
-           refind.append(j);
-        }
+        if (m_pFiffInfo->ch_names[j].indexOf("EEG001") >= 0) refind.append(j);
+
+        if (m_pFiffInfo->ch_names[j].indexOf("EEG002") >= 0) refind.append(j);
+
+        if (m_pFiffInfo->ch_names[j].indexOf("EEG024") >= 0) refind.append(j);
+
+        if (m_pFiffInfo->ch_names[j].indexOf("EEG025") >= 0) refind.append(j);
     }
```
Comment (chdinh) [inline applications/mne_x/plugins/rthpi/rthpi.cpp]: are you using the eeg channels for the hpi tracking?

File: applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h [hunk 4]
```diff
@@ -42,7 +42,7 @@
 // INCLUDES
 //=============================================================================================================
 
-#include "../ui_selectionmanagerwindow.h"
+#include "ui_selectionmanagerwindow.h"
```
Comment (chdinh) [inline applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h]: is this correct?

File: MNE/utils/mp/adaptivemp.cpp (hunk 1 above)
Comment (liminsun) [inline MNE/utils/mp/adaptivemp.cpp]: >> @@ -209,7 +209,7 @@ QList<QList<GaborAtom> >
>> AdaptiveMp::matching_pursuit(MatrixXd signal, qint32 ma
>>          k = 0;
>>          s = sample_count;
>>          p = floor(sample_count / 2);
>> -        j = floor(log2(sample_count));//log(sample_count) / log(2));
>> +        j = floor(log10(sample_count));//log(sample_count) / log(2));
>
> Is there a specific reason to change this line?
No. Just for passing compiler.

> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/mne-tools/mne-cpp/pull/170/files#r26209195



The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

File: applications/mne_x/plugins/babymeg/FormFiles/babymeghpidgl.cpp (hunk 2 above)
Comment (liminsun) [inline applications/mne_x/plugins/babymeg/FormFiles/babymeghpidgl.cpp]: >> +
>> +babymeghpidgl::~babymeghpidgl()
>> +{
>> +    delete ui;
>> +}
>> +
>> +void babymeghpidgl::CancelProc()
>> +{
>> +
>> +}
>> +
>> +void babymeghpidgl::OKProc(QAbstractButton *b)
>> +{
>> +    qDebug()<<"Clicked group button";
>> +
>> +    auto s(b->text());
>
> please don't use the auto variable

Ok. Please remove auto.




The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

File: applications/mne_x/plugins/rthpi/rthpi.cpp (hunk 3 above)
Comment (liminsun) [inline applications/mne_x/plugins/rthpi/rthpi.cpp]: >> -        if (m_pFiffInfo->ch_names[j].indexOf("TRG_014") >= 0) {
>> -           refind.append(j);
>> -        }
>> -        if (m_pFiffInfo->ch_names[j].indexOf("TRG_015") >= 0) {
>> -           refind.append(j);
>> -        }
>> -        if (m_pFiffInfo->ch_names[j].indexOf("TRG_016") >= 0) {
>> -           refind.append(j);
>> -        }
>> +        if (m_pFiffInfo->ch_names[j].indexOf("EEG001") >= 0)
>> refind.append(j);
>> +
>> +        if (m_pFiffInfo->ch_names[j].indexOf("EEG002") >= 0)
>> refind.append(j);
>> +
>> +        if (m_pFiffInfo->ch_names[j].indexOf("EEG024") >= 0)
>> refind.append(j);
>> +
>> +        if (m_pFiffInfo->ch_names[j].indexOf("EEG025") >= 0)
>> refind.append(j);
>>      }
>
> are you using the eeg channels for the hpi tracking?
Currently yes. Later we will change to the trigger lines. Right now, the
cables are not ready.




The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

File: applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h (hunk 4 above)
Comment (liminsun) [inline applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h]: >> @@ -42,7 +42,7 @@
>>  // INCLUDES
>>  //=============================================================================================================
>>
>> -#include "../ui_selectionmanagerwindow.h"
>> +#include "ui_selectionmanagerwindow.h"
>
> is this correct?

I did not chang this part. Maybe Chiran does that. However, I think it
belongs to compiling issue.

Also, chiran's rthpi plugin might have problems. We met crash when using
his plugin. But this should not be serious. Let him check his code first.






The information in this e-mail is intended only for the person to whom it is
addressed. If you believe this e-mail was sent to you in error and the e-mail
contains patient information, please contact the Partners Compliance HelpLine at
http://www.partners.org/complianceline . If the e-mail was sent to you in error
but does not contain patient information, please contact the sender and properly
dispose of the e-mail.

File: MNE/utils/mp/adaptivemp.cpp (hunk 1 above)
Comment (chdinh) [inline MNE/utils/mp/adaptivemp.cpp]: Okay than we should definitvely divide by log10(2), i.e., log10(sample_count) /  log10(2), to still compute the logarithmus dualis

File: applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h (hunk 4 above)
Comment (chdinh) [inline applications/mne_x_libs/xDisp/helpers/selectionmanagerwindow.h]: alright

**Pull request merged**